(DOCS-8698) Add AWS account-level logs guide (#25398)
DataDog/documentation: content/en/logs/guide/aws-account-level-logs.md

---
title: AWS Account-Level Log Subscriptions
further_reading:
- link: "/logs/explorer/"
  tag: "Documentation"
  text: "Learn how to explore your logs"
---

> You can use account-level log subscriptions in your AWS environment to automatically forward all of your CloudWatch logs to Datadog. With an account-level log subscription, you don't need to manually configure log forwarding when you have a new log source, or when AWS releases a new service. You can also define your own selection criteria or filter pattern for granular control over which logs are forwarded.

Reviewer: ```suggestion
## Overview

You can use account-level log subscriptions in your AWS environment to automatically forward all of your CloudWatch logs to Datadog. With an account-level log subscription, you don't need to manually configure log forwarding when you have a new log source, or when AWS releases a new service. You can also define your own selection criteria or filter pattern, for more control over which logs are forwarded.
```

## Create an account-level log subscription

> For the simplest setup, use [CloudFormation](#cloudformation-recommended) to create an Amazon Data Firehose and associated resources in each of your selected regions.

Reviewer: ```suggestion
There are two ways to create an account-level log subscription, through [CloudFormation](#cloudformation-recommended) and through [manual setup](#manual). For the simplest setup, use CloudFormation to create an Amazon Data Firehose and associated resources in each of your selected regions.
```

### CloudFormation (recommended)

1. Copy the URL of the CloudFormation template:

{{< code-block lang="bash" filename="" disable_copy="false" >}}
https://datadog-cloudformation-template.s3.amazonaws.com/aws_account_level_logs/main.yaml
{{< /code-block >}}

2. Go to [CloudFormation][1] in the AWS console.
3. Click **Create stack**.
    - Select `With new resources (standard)`.
4. Leave the options to **Choose an existing template** and **Amazon S3 URL** checked.
5. In the **Amazon S3 URL** field, paste the URL of the CloudFormation template.
6. Click **Next**.
7. In the **Stack name** field, provide a descriptive name such as `datadog-account-level-logs-stack`.
8. In the **ApiKey** field, paste a valid [Datadog API key][4] value.

> 9. In the **Regions** field, enter a comma-separated list of regions to enable metric streaming.

Reviewer 2: I would change `regions` to `AWS Regions` so it's more clear what regions we're looking for.
Also, `enable metric streaming` should be something like `enable account-level logs subscription"

Author: Thanks for reviewing and catching that, updated in [c7373f1](https://github.com/DataDog/documentation/pull/25398/commits/c7373f188698e361872e2921903a94d1665d026e)

	--statement-id "forwarder-function" \
	--principal "logs.amazonaws.com" \
	--action "lambda:InvokeFunction" \
	--source-arn "arn:aws:logs:<REGION>:<ACCOUNT_ID>:log-group:*" \
	--source-account "<ACCOUNT_ID>"
```

3. Create an account-level subscription filter policy. In the example provided below, all log events that contain the string `ERROR` are streamed, except those in the log groups named `LogGroupToExclude1` and `LogGroupToExclude2`.
   - Replace `FORWARDER_ARN` with the ARN of the Datadog Forwarder Lambda function.

```bash
aws logs put-account-policy \
  --policy-name "ExamplePolicyLambda" \
  --policy-type "SUBSCRIPTION_FILTER_POLICY" \
  --policy-document '{"DestinationArn":"<FORWARDER_ARN>", "FilterPattern": "", "Distribution": "Random"}' \
  --scope "ALL"
```

**Note**: To exclude certain log groups from log forwarding, use the `--selection-criteria` option as outlined in the [command reference][103].

[101]: /logs/guide/send-aws-services-logs-with-the-datadog-lambda-function/
[102]: https://aws.amazon.com/cli/
[103]: https://docs.aws.amazon.com/cli/latest/reference/logs/put-account-policy.html
{{% /tab %}}
{{% tab "Amazon Data Firehose" %}}

> #### Create S3 bucket and role for Amazon Data Firehose

Reviewer: ```suggestion
#### Create an S3 bucket and a role for Amazon Data Firehose
```

The following steps guide you through creating a bucket and IAM role. This role grants Amazon Data Firehose permission to put data into your Amazon S3 bucket in case of delivery failures.

1. Use the [AWS CLI][201] to create an S3 Bucket. Optionally, you can use an existing bucket.
   - Replace `<BUCKET_NAME>` with the name for your S3 bucket.
   - Replace `<REGION>` with the region for your S3 bucket.

```
aws s3api create-bucket \
  --bucket MY-BUCKET \
  --create-bucket-configuration LocationConstraint=<REGION>
```

2. Create a `TrustPolicyForFirehose.json` file with the following statement:

```bash
{
  "Statement": {
    "Effect": "Allow",
    "Principal": { "Service": "firehose.amazonaws.com" },
    "Action": "sts:AssumeRole"
    } 
}
```

3. Create an IAM role, specifying the trust policy file:

```bash
aws iam create-role \
  --role-name FirehosetoS3Role \
  --assume-role-policy-document file://./TrustPolicyForFirehose.json
```

>    **Note**: The returned **Role.Arn** value is used in a later step.

Reviewer: Recommend moving up:
Create an IAM role, specifying the trust policy file. **Note**: The returned **Role.Arn** value is used in a later step.
